src/ui/rcb-ui.svelte

```diff
@@ -102,25 +102,16 @@
     }
 
     purchaseOperationHelper
-      .startPurchase(
+      .checkoutStart(
         appUserId,
         productId,
         purchaseOptionToUse,
-        customerEmail,
         rcPackage.webBillingProduct.presentedOfferingContext,
       )
       .then((result) => {
-        if (result.next_action === "collect_payment_info") {
-          lastError = null;
-          currentView = "needs-payment-info";
-          paymentInfoCollectionMetadata = result;
-          return;
-        }
-        if (result.next_action === "completed") {
-          lastError = null;
-          currentView = "success";
-          return;
-        }
```

Comment:
This was unreachable code, since `next_action` is always set to `collect_payment_info` by the backend

Reply:
That's correct Nicola, the old endpoints (`/subscribe` and `/purchase` endpoints) are kept untouched and will return the field `collect_payment_info`.

I tested all of this in a canary using the published version of the SDK before shipping the BE changes.

Reply:
My question was related to the `completed` value.

When was that value returned?
In what scenario?

I understand that "recently" we updated to always return `collect_payment_info` my question is:

Did we break older version of the sdk that supported `completed` when we stopped sending it?

Reply:
As discussed in the 1:1 I was more worried about the potential issue happening due to the backend changes more than this change. Sorry for the confusion.